NuGet.Services.Dashboard.Operations/Infrastructure/CommandLineParser.cs

```diff
@@ -70,7 +72,16 @@ public void ExtractOptions(ICommand command, IEnumerator<string> argsEnumerator)
                     throw new CommandLineException(TaskResources.MissingOptionValueError, option);
                 }
 
-                AssignValue(command, propInfo, option, value);
+                string newValue;
+
+                if (SecretConfigurationManager.TryInjectSecret(value, out newValue))
```

Comment:
You are correct in that the keyvault retrieval is nontrivial overhead.

The implmentation of injectAsync ([https://github.com/NuGet/NuGet.Jobs/blob/master/src/NuGet.Services.KeyVault/SecretInjector.cs](https://github.com/NuGet/NuGet.Jobs/blob/master/src/NuGet.Services.KeyVault/SecretInjector.cs)) already only tried to retrieve if the secret is framed ("$$" by default).

---
In reply to: [71936820](https://github.com/NuGet/NuGet.Services.Dashboard/pull/187#discussion_r71936820) [](ancestors = 71936820)